packages/graphql-server-hapi/src/hapiApollo.ts

```diff
@@ -71,23 +77,44 @@ graphqlHapi.attributes = {
   version: '0.0.1',
 };
 
-function verifyPayload(payload, reply) {
-  if (!payload) {
-    return reply(createErr(500, 'POST body missing.'));
-  }
+function assignBuffer(method, payload, query, reply) {
+    switch ( method ) {
+        case 'get':
+            if (!query) {
+                return reply(createErr(500, 'GET query missing.'));
+            }
+            return reply(query);
+        case 'post':
+            if (!payload) {
+                return reply(createErr(500, 'POST body missing.'));
+            }
+            return reply(payload);
+        default:
+            return reply(createErr(405, 'Apollo Server supports only GET/POST requests.'));
+    }
+}
 
+function assignIsBatch(method, buffer, reply) {
   // TODO: do something different here if the body is an array.
   // Throw an error if body isn't either array or object.
-  reply(payload && Array.isArray(payload));
+
+  switch ( method ) {
+      case 'get':
+          return reply(false);
```

Comment:
Yeah, but I don't think monstrous URLs should be our problem. There's a real risk for that already with normal queries. Most servers don't support URLs longer than 2000 characters, and I think even the introspection query might be longer than that. But that's really not something we should worry about. Maybe we can just put a warning in the README.